spec/support/shared_examples.rb

```diff
@@ -0,0 +1,35 @@
+shared_examples 'a patcher' do
+  let(:patcher) { described_class.new(patch_location) }
+  let(:patch_location) { nil }
+
+  describe '#mutate_schema' do
+    it 'implements mutate schema' do
+      expect(patcher).to respond_to(:mutate_schema)
+    end
+  end
+
+  describe '#location' do
+    let(:patch_location) { nil }
```

Comment:
We could wrap this in `context 'no location is given' do`. Then when we run the specs with `-fd` the behaviour becomes obvious (compare with what it outputs now). Yes, I'm `context`-crazy.